presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -176,12 +176,23 @@ public PlanWithProperties visitAggregation(AggregationNode node, HashComputation
                             child.getNode(),
                             node.getAggregations(),
                             node.getGroupingSets(),
+                            node.getPreGroupedSymbols(),
                             node.getStep(),
                             hashSymbol,
                             node.getGroupIdSymbol()),
                     hashSymbol.isPresent() ? ImmutableMap.of(groupByHash.get(), hashSymbol.get()) : ImmutableMap.of());
         }
 
+        private boolean canSkipHashGeneration(AggregationNode node)
+        {
+            if (!node.getPreGroupedSymbols().isEmpty() &&
+                    node.getGroupingSets().size() == 1
```

Comment:
add `AggregationNode#isStreamingAggregation` method. Use it in local execution planner too